### DISCUSSION TITLE
pLDDT values in the downloaded pdb file

### DISCUSSION THREAD

**Discussion by angel-herraez** [facebookresearch/esm#608] [ESM Metagenomic Atlas]
Hello. I am testing the online "Fold sequence" tool with the idea of introducing structure prediction to my students. Big thanks for offering this easy interface.
I enter a sequence and get the 3D display. Then I download the pdb file. I thought the pLDDT values might be in the B-factor column. However, when I open the file (in Jmol) and color "by temperature" (i.e. according to B-factor), the most reliable parts do not match what I see in the colored display in the web folding tool.
¿How can I retrieve the pLDDT values in a way that matches what I see in the Fold interface?
Thanks for any hints

Example for GECEQCFSDGGDCTTCFNNGTGPCANCLAGYPAGCSNSDCTAFLSQCYGGC :
![imagen](https://github.com/facebookresearch/esm/assets/49884385/742e1193-6e35-4343-9b0a-60dbe7d4068b)

**Comment by Basso42** [comment]
Really interested as well, is there any way to recover pLDDT values from the .pdb file?

**Comment by angel-herraez** [comment]
Basso, I'm so glad that someone also shows interest on this. It's a 
shortcoming to have such nice tool available and not be able to save the 
result as it is displayed.

In the meantime, I researched a little and found inspiration to 
interpret the output. A first solution is here
https://biomodel.uah.es/Jmol/fold_AI/

I will keep improving that.

·
  Dr. Angel Herráez
  Biochemistry and Molecular Biology,
  Dept. of Systems Biology, University of Alcalá
  E-28805 Alcalá de Henares  (Madrid), Spain
  https://biomodel.uah.es/

**Comment by angel-herraez** [comment]
Further improvements. Now you can:

1. Color the structure according to the pLDDT data in the file from ESM.
2. Save an image (PNG) or an image+script to reproduce the rendering later in Jmol/JSmol (PNGJ).
3. Save to a new pdb file that has the pLDDT values converted into standard PDB B-factor ("temperature") values.
4. Retrieve a text with all atom IDs and their pLDDT values, ready for pasting into e.g. a spreadsheet.

**Comment by Basso42** [comment]
Hello Angel, thank for your answers ; unfortunately I think that the beta factors do not represent the pLDDT value in the ESM Fold model. Indeed when you compute, atom by atom the average beta factor of an amino acid it does not match the one displayed by the ESM Atlas visual interface.

**Reply by angel-herraez** [reply]
You don't think so? See this example, obtained with sequence:
MALWMRLLPLLALLALWGPDPAAAFVNQHLCGSHLVEALYLVCGERGFFYTPKTRREAEDLQVGQVELGGGPGAGSLQPLALEGSLQKRGIVEQCCTSICSLYQLENYCN
Submitted to https://esmatlas.com/resources?action=fold
and the screenshot + download the pdb file and color it using https://biomodel.uah.es/Jmol/fold_AI/EMSfold.htm 
![imagen](https://github.com/facebookresearch/esm/assets/49884385/df0f6a36-4b3a-4c1f-9c75-6f339062311f)
![imagen](https://github.com/facebookresearch/esm/assets/49884385/701dd884-98af-48d8-b1c5-5ffb8d6db38d)
They look quite similar to me.
It is true that my coloring is based just on the alpha carbons pLDDT value, and not on the average of all atoms in eh residue. That may explain the small differences in some points.  I will try to implement that.

**Reply by angel-herraez** [reply]
Yes, I think we are in the correct situation. This is quick test using the average B within each residue rather than the b for the alpha carbon:
![imagen](https://github.com/facebookresearch/esm/assets/49884385/9121195d-3f6f-4d84-989d-01f9341592c3)

**Reply by Basso42** [reply]
Pardon me, you seem right ; I was confused because the average B factor per amino acid does not match the pLDDT of the same amino acid on the website.  See here for the second amino acid of a sequence I retrieved from ESM Fold
![Screenshot from 2023-12-12 10-21-30](https://github.com/facebookresearch/esm/assets/91699102/8ed40488-534b-4702-ba50-c069d625506d)
![Screenshot from 2023-12-12 10-24-25](https://github.com/facebookresearch/esm/assets/91699102/a2f74b16-2bff-4a8f-b211-9617a57cc152)
I don't see any reason why it should be different ?

Thank you again for the tool you developped!
